dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java

```diff
@@ -534,10 +534,23 @@ public WebSphereBeanDeploymentArchive getBeanDeploymentArchiveFromClass(Class<?>
 
         if (wbda == null) {
             for (WebSphereBeanDeploymentArchive bda : orderedBDAs) {
-                if ((bda.getAllClazzes().containsKey(clazz.getName())) && (bda.getAllClazzes().containsValue(clazz))) {
-                    wbda = bda;
-                    classBDAMap.put(clazz, bda);
-                    break;
+                if (bda.getAllClazzes().contains(clazz.getName())) {
+
+                    Class<?> bdaClazz = null;
+                    try {
+                        bdaClazz = CDIUtils.loadClass(bda.getClassLoader(), clazz.getName());
```

Comment:
Although you are asking the BDA's classloader to load the class, that might not be the same classloader that actually does load it. If it isn't then you probably shouldn't add it ... or is the `if (bdaClazz == clazz) {` enough here??